shared/utils/git_cache.py

```
@@ -0,0 +1,359 @@
+        return False, error_msg
+    except subprocess.TimeoutExpired:
+        error_msg = f"Timeout creating cache repository after 300 seconds"
+        logger.error(error_msg)
+        return False, error_msg
+    except Exception as e:
+        logger.error(f"Unexpected error creating cache: {e}")
+        return False, str(e)
+
+
+def update_cache_repo(
+    cache_path: str, auth_url: str = None
```

Ruff (0.14.13)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)